app/routes.py

```diff
@@ -0,0 +1,357 @@
+    return jsonify(customers_response), 200
+
+@customers_bp.route("/<customer_id>", methods=["GET"])
+def get_customer(customer_id):
+    if customer_id.isnumeric() != True:
+        return {"message": "Customer id provided is not a number."}, 400
+    
+    customer = Customer.query.get(customer_id)
+
+    if customer is None:
+        return {"message": f"Customer {customer_id} was not found"}, 404
+
+    return {
+        "id" : customer.id,
+        "name" : customer.name,
+        "postal_code" : customer.postal_code,
+        "phone" : customer.phone
+    }
+@customers_bp.route("/<customer_id>", methods=["PUT"])
+def update_customer(customer_id):
+    customer = Customer.query.get(customer_id)
+    request_body = request.get_json()
+    
+    if customer is None:
+        return {"message": f"Customer {customer_id} was not found"}, 404
+
+    if "name" not in request_body or "postal_code" not in request_body or "phone" not in request_body:
+        return {
+        "details": "Invalid data"
+        }, 400
+        
+    customer.name = request_body["name"]
+    customer.phone = request_body["phone"]
+    customer.postal_code = request_body["postal_code"]
+    
+    db.session.commit()
+
+    return {
+        "id" : customer.id,
+        "name" : customer.name,
+        "postal_code" : customer.postal_code,
+        "phone" : customer.phone,
+        "registered_at" : datetime.now()
+    }
+
+@customers_bp.route("/<customer_id>", methods=["DELETE"])
+def delete_customer(customer_id):
+
+    customer = Customer.query.get(customer_id)
+
+    if customer is None:
+        return {"message": f"Customer {customer_id} was not found"}, 404
+
+    db.session.delete(customer)
+    db.session.commit()
+
+    return jsonify({"id": customer.id}), 200
+##Rental code
+rentals_bp = Blueprint("rentals", __name__, url_prefix="/rentals")
+
+@rentals_bp.route("/check-out", methods = ["POST"])
+def rental_check_out():
+    rental_request = request.get_json()
+
+    if "customer_id" not in rental_request or "video_id" not in rental_request:
+        return jsonify(details="Request body must include customer id and video id"), 400
+
+    customer_id = rental_request["customer_id"]
+    video_id = rental_request["video_id"]
+
+    try: 
+        video_id = int(video_id)
+        customer_id = int(customer_id)
+    except:
+        return jsonify(None), 400
+
+    customer = Customer.query.get(customer_id)
+    video = Video.query.get(video_id)
+    if customer is None or video is None:
+        return jsonify(None), 404
+
+    num_currently_checked_out = Rental.query.filter_by(video_id=video.id, checked_out=True).count()
+    available_inventory = video.total_inventory - num_currently_checked_out
+
+    if available_inventory == 0:
+        return jsonify(message="Could not perform checkout"), 400
+
+    new_rental_check_out= Rental(
+        customer_id=customer.id,
+        video_id=video.id,
+        due_date=(datetime.now() + timedelta(days=7)),
```

Comment:
Consider using `utcnow()` rather than `now()`, especially if there's no timezone information on the datetime in the database. When working with dates and time, it's usually a good idea to pass them around and store them in UTC, and only convert them to local time when needed for display (like in a UI). Working with dates and times is tricky!